File: frontend/src/api/founderDashboard.ts
import apiClient from './client'

export interface DepartmentSummary {
  department_id: number
  department_name: string
  budget_planned: number
  budget_actual: number
  budget_remaining: number
  budget_execution_percent: number
  expenses_count: number
  expenses_pending: number
  expenses_paid_amount: number
  employees_count: number
  payroll_planned: number
  payroll_actual: number
  avg_kpi_achievement: number | null
}

export interface CompanySummary {
  total_budget_planned: number
  total_budget_actual: number
  total_budget_remaining: number
  total_budget_execution_percent: number
  total_expenses_count: number
  total_expenses_pending: number
  total_expenses_paid: number
  total_employees: number
  total_payroll_planned: number
  total_payroll_actual: number
  departments_count: number
}

export interface TopCategoryByDepartment {
  department_id: number
  department_name: string
  category_id: number | null
  category_name: string
  category_type: string | null
  amount: number
  execution_percent: number
}

export interface ExpenseTrend {
  month: number
  month_name: string
  planned: number
  actual: number
  execution_percent: number
}

export interface DepartmentKPI {
  department_id: number
  department_name: string
  avg_achievement: number
  employees_with_kpi: number
  total_employees: number
  coverage_percent: number
}

export interface BudgetAlert {
  alert_type: string
  department_id: number
  department_name: string
  category_id: number | null
  category_name: string | null
  planned: number
  actual: number
  execution_percent: number
  message: string
}

export interface FounderDashboardData {
  year: number
  month: number | null
  company_summary: CompanySummary
  departments: DepartmentSummary[]
  top_categories_by_department: TopCategoryByDepartment[]
  expense_trends: ExpenseTrend[]
  department_kpis: DepartmentKPI[]
  alerts: BudgetAlert[]
}

export interface FounderDepartmentDetails {
  department_id: number
  department_name: string
  year: number
  month: number | null
  budget_by_category: any[]
  expenses_by_month: any[]
  employee_kpis: any[]
  recent_expenses: any[]
}

export const founderDashboardApi = {
  getDashboard: async (params?: { year?: number; month?: number }): Promise<FounderDashboardData> => {
    const { data } = await apiClient.get('founder/dashboard', { params })
    return data
  },

  getDepartmentDetails: async (
    departmentId: number,
    params?: { year?: number; month?: number }
  ): Promise<FounderDepartmentDetails> => {
    const { data } = await apiClient.get(`founder/department/${departmentId}`, { params })
    return data
  },
}

File: backend/app/schemas/founder_dashboard.py
"""Pydantic schemas for Founder Dashboard"""
from typing import List, Optional
from decimal import Decimal
from pydantic import BaseModel


class DepartmentSummary(BaseModel):
    """Сводка по отделу"""
    department_id: int
    department_name: str
    # Бюджет
    budget_planned: Decimal
    budget_actual: Decimal
    budget_remaining: Decimal
    budget_execution_percent: float
    # Расходы
    expenses_count: int
    expenses_pending: int
    expenses_paid_amount: Decimal
    # Сотрудники и зарплаты
    employees_count: int
    payroll_planned: Decimal
    payroll_actual: Decimal
    # КПИ
    avg_kpi_achievement: Optional[float] = None


class CompanySummary(BaseModel):
    """Общая сводка по компании"""
    # Бюджет
    total_budget_planned: Decimal
    total_budget_actual: Decimal
    total_budget_remaining: Decimal
    total_budget_execution_percent: float
    # Расходы
    total_expenses_count: int
    total_expenses_pending: int
    total_expenses_paid: Decimal
    # Сотрудники
    total_employees: int
    total_payroll_planned: Decimal
    total_payroll_actual: Decimal
    # Отделы
    departments_count: int


class TopCategoryByDepartment(BaseModel):
    """Топ категория расходов по отделу"""
    department_id: int
    department_name: str
    category_id: Optional[int]
    category_name: str
    category_type: Optional[str]
    amount: Decimal
    execution_percent: float


class ExpenseTrend(BaseModel):
    """Тренд расходов по месяцам"""
    month: int
    month_name: str
    planned: Decimal
    actual: Decimal
    execution_percent: float


class DepartmentKPI(BaseModel):
    """КПИ отдела"""
    department_id: int
    department_name: str
    avg_achievement: float
    employees_with_kpi: int
    total_employees: int
    coverage_percent: float


class BudgetAlert(BaseModel):
    """Предупреждение о бюджете"""
    alert_type: str  # "overbudget", "high_utilization", "low_utilization"
    department_id: int
    department_name: str
    category_id: Optional[int]
    category_name: Optional[str]
    planned: Decimal
    actual: Decimal
    execution_percent: float
    message: str


class FounderDashboardData(BaseModel):
    """Главный дашборд учредителя"""
    year: int
    month: Optional[int] = None

    # Общая сводка
    company_summary: CompanySummary

    # Детали по отделам
    departments: List[DepartmentSummary]

    # Топ категории расходов по отделам
    top_categories_by_department: List[TopCategoryByDepartment]

    # Тренды
    expense_trends: List[ExpenseTrend]

    # КПИ
    department_kpis: List[DepartmentKPI]

    # Предупреждения
    alerts: List[BudgetAlert]


class FounderDepartmentDetails(BaseModel):
    """Детальная информация по отделу для учредителя"""
    department_id: int
    department_name: str
    year: int
    month: Optional[int] = None

    # Бюджет по категориям
    budget_by_category: List[dict]

    # Расходы по месяцам
    expenses_by_month: List[dict]

    # КПИ сотрудников
    employee_kpis: List[dict]

    # Последние расходы
    recent_expenses: List[dict]
